nupic/research/TP_shim.py

```diff
@@ -27,6 +27,9 @@
 import numpy
 
 from nupic.research.temporal_memory import TemporalMemory
+from nupic.research.monitor_mixin.temporal_memory_monitor_mixin import (
+  TemporalMemoryMonitorMixin)
+class MonitoredTemporalMemory(TemporalMemoryMonitorMixin, TemporalMemory): pass
```

Comment:
Good idea, I can put it in `nupic.research.monitor_mixin.temporal_memory_monitor_mixin`. I'll do that in a separate PR.